README.md

```diff
@@ -58,81 +54,52 @@ Include this badge in your readme if you make a new module that implements inter
 
 A valid (read: that follows this interface) IPLD format implementation the following API.
 
-### IPLD format utils
+IPLD Format APIs are restricted to a single IPLD Node, they never access any linked IPLD Nodes.
 
-#### `util.serialize(dagNode, callback)`
 
-> serializes a dagNode of an IPLD format into its binary format
+### `serialize(IpldNode)`
 
-`callback` must have the signature `function (err, binaryBlob)`, where `err` is an Error is the function fails and `binaryBlob` is a Buffer containing the serialized version.
+> Serializes the internal representation of an IPLD Node into a binary blob.
 
-#### `util.deserialize(binaryBlob, callback)`
+`IpldNode` is a previously deserialized binary Block.
 
-> deserializes a binary blob into the instance
+Returns a Promise containing a `Buffer` with the serialized version of the given IPLD Node.
 
-`callback` must have the signature `function (err, dagNode)`, where `err` is an Error if the function fails and `dagNode` is the dagNode that got deserialized in the process.
 
-#### `util.cid(binaryBlob[, options], callback)`
+### `deserialize(binaryBlob)`
 
-> get the CID of a binary blob
+> Deserialize into internal representation.
 
-Options include:
-  - version - the CID version to be used (defaults to 1)
-  - hashAlg - the hash algorithm to be used (default to the one set by the format)
+The result is a JavaScript object, which may also be a Proxy object in case the data shouldn’t be deserialized as a whole. Its fields are the public API that can be resolved through. It’s up to the format to add convenient methods for manipulating the data. The returned object may also be a Proxy object in case the data shouldn’t be deserialized as a whole.
 
-`callback` must have the signature `function (err, cid)`, where `err` is an Error if the function fails and `cid` is a CID instance of the binary blob.
+Returns a Promise containing the Javascript object. This object must be able to be serialized with a `serialize()` call.
 
-### Local resolver methods
 
-#### `resolver.resolve(binaryBlob, path, callback)`
+### `cid(binaryBlob, [options])`
 
-> resolves a path in block, returns the value and or a link and the partial missing path. This way the IPLD Resolver can fetch the link and continue to resolve.
+> Return the CID of the binary blob.
```

Comment:
This is more of a question.

From what I can tell implementation of the `cid` is pretty much the same across the board. Is there reason to require implementation of `cid` in first place ? Could this be made optional, so only implementation that need a custom behavior can provide it ?

Reply:
An IPLD Format has all the information it needs to calculate the CID. The multicodec code as well as the hash. So you can call `cid()` from layers that build on top of IPLD Formats without knowing those details.